logging/slog_logger.go

```diff
@@ -1,75 +1,119 @@
-	if env == Production {
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else if env == Development {
-		// Even though code is same for both environments, we are keeping it separate to show that we can have different
-		// implementations for different environments. For example, we can have log levels set differently
-		logHandler := slog.NewJSONHandler(os.Stdout, nil).WithAttrs([]slog.Attr{slog.String("env", string(env))})
-		logger := slog.New(logHandler)
-		return &SLogger{
-			logger: logger,
-		}
-	} else {
-		panic(fmt.Sprintf("Unknown environment. Expected %s or %s. Received %s.", Development, Production, env))
+// NewSlogTextLogger creates a new SLogger with a text handler
+//
+//	outputWriter is the writer to write the logs to (typically os.Stdout,
+//	  but can also use a io.MultiWriter(os.Stdout, file) to write to multiple outputs)
+//	logLevel is the minimum level to log
+//	logSource if true, adds source information to the log
+func NewSlogTextLogger(outputWriter io.Writer, logLevel slog.Level, logSource bool) *SLogger {
+	handler := slog.NewTextHandler(outputWriter, &slog.HandlerOptions{
+		Level:     logLevel, // prints debug and above
+		AddSource: logSource,
+	})
+	logger := slog.New(handler)
+	return &SLogger{
+		logger,
+	}
+}
+
+// NewSlogJsonLogger creates a new SLogger with a Json handler
+//
+//	outputWriter is the writer to write the logs to (typically os.Stdout,
+//	  but can also use a io.MultiWriter(os.Stdout, file) to write to multiple outputs)
+//	logLevel is the minimum level to log
+//	logSource if true, adds source information to the log
+func NewSlogJsonLogger(outputWriter io.Writer, logLevel slog.Level, logSource bool) *SLogger {
+	handler := slog.NewJSONHandler(outputWriter, &slog.HandlerOptions{
+		Level:     logLevel, // prints debug and above
+		AddSource: logSource,
+	})
+	logger := slog.New(handler)
+	return &SLogger{
+		logger,
 	}
 }
 
 func (s SLogger) Debug(msg string, tags ...any) {
-	s.logger.Debug(msg, tags...)
+	s.logCorrectSource(slog.LevelDebug, msg, tags...)
 }
 
 func (s SLogger) Info(msg string, tags ...any) {
-	s.logger.Info(msg, tags...)
+	s.logCorrectSource(slog.LevelInfo, msg, tags...)
 }
 
 func (s SLogger) Warn(msg string, tags ...any) {
-	s.logger.Warn(msg, tags...)
+	s.logCorrectSource(slog.LevelWarn, msg, tags...)
 }
 
 func (s SLogger) Error(msg string, tags ...any) {
-	s.logger.Error(msg, tags...)
+	s.logCorrectSource(slog.LevelError, msg, tags...)
 }
 
 func (s SLogger) Fatal(msg string, tags ...any) {
-	s.logger.Error(msg, tags...)
+	s.logCorrectSource(slog.LevelError, msg, tags...)
 	os.Exit(1)
 }
 
+// logCorrectSource logs the message with the correct source information
+// adapted from https://pkg.go.dev/log/slog#example-package-Wrapping
+func (s SLogger) logCorrectSource(level slog.Level, msg string, tags ...any) {
```

Comment:
do we always want this feature?
should this be an option the user configures?

Reply:
I think it is an user option - https://github.com/Layr-Labs/eigensdk-go/pull/127/files#diff-54eaa5e5eaaf4b31c21b1f44503ed097a88564b92a04a4f35a17ce5240612b6cR52 ?

Reply:
Indeed. resolving